src/App.tsx

```diff
@@ -1,24 +1,312 @@
+import { Filter } from './components/Filter';
+import { ShowTodos } from './types/ShowTodos';
+import { ShowError } from './types/ShowErrors';
 
-const USER_ID = 0;
+const USER_ID = 10565;
+
+const todosFromServer = client
+  .get<Todo[]>(`/todos?userId=${USER_ID}`);
+const createTodoOnServer = (title: string) => client
+  .post<Todo>('/todos', {
+  title,
+  userId: USER_ID,
+  completed: false,
+});
+const deleteTodoOnServer = (todoId: number) => client
+  .delete(`/todos/${todoId}`);
+const toggleTodoOnServer = (todoId: number, completed: boolean) => client
+  .patch<Todo>(`/todos/${todoId}`, { completed: !completed });
+const updateTodoOnServer = (todoId: number, title: string) => client
+  .patch<Todo>(`/todos/${todoId}`, { title });
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<ShowError | null>(null);
+  const [selectedTodos, setSelectedTodos] = useState<ShowTodos>(ShowTodos.All);
+  const [todoTitle, setTodoTitle] = useState<string | null>(null);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const clearError = () => {
+    setTimeout(() => {
+      setError(null);
+    }, 3000);
+  };
+
+  const showErrors = (showError: ShowError) => {
+    switch (showError) {
+      case ShowError.fetchTodos:
+        setError(ShowError.fetchTodos);
+        break;
+      case ShowError.createTodo:
+        setError(ShowError.createTodo);
+        break;
+      case ShowError.addTodo:
+        setError(ShowError.addTodo);
+        break;
+      case ShowError.deleteTodo:
+        setError(ShowError.deleteTodo);
+        break;
+      case ShowError.updateTodo:
+        setError(ShowError.updateTodo);
+        break;
+
+      default:
+        break;
+    }
+
+    clearError();
+  };
+
+  const handleFilterSelect = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    switch (event.currentTarget.textContent) {
+      case 'Active':
+        setSelectedTodos(ShowTodos.Active);
+        break;
+      case 'Completed':
+        setSelectedTodos(ShowTodos.Completed);
+        break;
+      default:
+        setSelectedTodos(ShowTodos.All);
+        break;
+    }
+  };
+
+  const getFilteredTodos = (showTodos: ShowTodos) => {
+    switch (showTodos) {
+      case ShowTodos.Active:
+        return todos.filter(todo => !todo.completed);
+      case ShowTodos.Completed:
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  };
+
+  useEffect(() => {
+    setError(null);
+    todosFromServer
+      .then(fetchedTodos => setTodos(fetchedTodos))
+      .catch(() => showErrors(ShowError.fetchTodos));
+  }, []);
+
+  const filteredTodos = React.useMemo(() => {
+    return getFilteredTodos(selectedTodos);
+  }, [selectedTodos, todos]);
+
+  const createTodo = (event: React.FormEvent) => {
+    if (todoTitle === '') {
```

Comment:
```suggestion
    if (!todoTitle) {
```

Reply:
Done, but I prefer to not use uses type conversion in boolean context.